pkg/util/stringutil/stringutil_test.go

```diff
@@ -0,0 +1,144 @@
+package stringutil
+
+import (
+	"fmt"
+	"math/rand"
+	"runtime"
+	"runtime/debug"
+	"sync"
+	"testing"
+)
+
+var oldBank sync.Map
+
+// This is the old implementation of the string bank to use for comparison benchmarks
+func BankLegacy(s string) string {
+	ss, _ := oldBank.LoadOrStore(s, s)
+	return ss.(string)
+}
+
+func ClearBankLegacy() {
+	oldBank = sync.Map{}
+}
```

Comment:
Should these be exported?

Reply:
Exported is fine in the test context, since `_test` files aren't compiled outside of testing, but not necessary. Probably worth a change.

Reply:
I didn't know that about compilation, always good to expand my knowledge of Go esoteria a bit.